src/unread/unreadPmsReducer.js

```diff
@@ -46,7 +47,32 @@ const eventUpdateMessageFlags = (state, action) => {
   if (action.op === 'add') {
     return removeItemsDeeply(state, action.messages);
   } else if (action.op === 'remove') {
-    // we do not support that operation
+    const { message_details } = action;
+    if (message_details === undefined) {
+      logging.warn('Got update_message_flags event without message_details.');
+      return state;
+    }
+
+    let newState = [...state];
```

Comment:
This copying of the array, and also the one at the `return` below, can be left out; we don't mutate the array anywhere.

(The `addItemsToPmArray` helper already avoids mutation.  That can be seen in its type: the first parameter, where we pass this `newState` to, is a `$ReadOnlyArray`.)

This data structure is never going to be a model of efficiency whatever we do with it -- the real fix is to replace it with an efficient data structure, like we've done for unread stream messages in `unreadModel.js`. But we can avoid extra copies where it's easy to do so.